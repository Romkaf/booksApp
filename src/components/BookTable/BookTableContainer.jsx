import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchBooks } from '@models/actions';
import BookTable from './BookTable';

const BookTableContainer = () => {
	const {
		books: { books, loading },
		filters,
		pagination,
	} = useSelector((state) => state);

	const dispatch = useDispatch();

	useEffect(() => dispatch(fetchBooks({ ...filters, ...pagination })), [
		filters,
		pagination,
	]);

	return <BookTable books={books} loading={loading} />;
};

export default BookTableContainer;
