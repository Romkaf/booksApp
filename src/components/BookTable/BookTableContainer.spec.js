import React from 'react';
import BookTableContainer from './BookTableContainer';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

describe('BookTableContainer', () => {
	const mockStore = configureMockStore([thunk]);

	const store = mockStore({
		books: { books: [] },
		pagination: { currentPage: 1, perPage: 5 },
		filters: { titleFilter: '', authorFilter: '' },
	});

	const wrapper = mount(
		<Provider store={store}>
			<BookTableContainer />
		</Provider>,
	);

	it('should render component', () => {
		const component = wrapper.find('BookTableContainer');
		expect(component).toMatchSnapshot();
	});
});
