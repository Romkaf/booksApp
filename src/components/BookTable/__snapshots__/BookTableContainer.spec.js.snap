// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookTableContainer should render component 1`] = `
<BookTableContainer>
  <Memo(BookTable)
    books={Array []}
  >
    <div
      className="wrapper"
    >
      <table
        className="root"
      >
        <Memo(TableHead)>
          <thead
            className="root"
          >
            <tr>
              <th
                className="ceil"
                key="Title"
              >
                Title
              </th>
              <th
                className="ceil"
                key="Authors"
              >
                Authors
              </th>
              <th
                className="ceil"
                key="Publisher"
              >
                Publisher
              </th>
              <th
                className="ceil"
                key="Date"
              >
                Date
              </th>
            </tr>
          </thead>
        </Memo(TableHead)>
        <TableBodyContainer
          books={Array []}
        >
          <TableBody
            books={Array []}
          >
            <tbody />
          </TableBody>
        </TableBodyContainer>
      </table>
      <PaginationContainer>
        <Pagination
          currentPage={1}
          onPageClick={[Function]}
          pages={Array []}
        >
          <ul
            className="root"
          />
        </Pagination>
      </PaginationContainer>
    </div>
  </Memo(BookTable)>
</BookTableContainer>
`;
